src/test/tests/graphics/scenery/tests/unit/backends/BufferUtilsTests.kt

```diff
@@ -0,0 +1,132 @@
+package graphics.scenery.tests.unit.backends
+
+import graphics.scenery.BufferUtils
+import graphics.scenery.utils.LazyLogger
+import org.junit.Test
+import java.nio.*
+import kotlin.test.assertEquals
+
+
+/**
+ * Tests for [BufferUtils].
+ *
+ * @author Aryaman Gupta <aryaman1994@gmail.com>
+ */
+
+class BufferUtilsTests {
+    private val logger by LazyLogger()
+
+    @Test
+    fun testAllocateFloat() {
+        logger.info("Testing allocation of a new direct float buffer ...")
+        val size = kotlin.random.Random.nextInt(0, 10000)
+        val floatBuf = BufferUtils.allocateFloat(size)
+
+        assertEquals(size, floatBuf.capacity(), "Float buffer capacity was expected to be $size, but is ${floatBuf.capacity()}")
+        assertEquals (ByteOrder.nativeOrder(), floatBuf.order(), "Float buffer was expected to be in ${ByteOrder.nativeOrder()}, but is ${floatBuf.order()}")
+    }
+
+    @Test
+    fun testAllocateFloatAndPut() {
+        logger.info("Testing allocation and initialization of a new direct float buffer with elements of a given array ...")
+        val size = kotlin.random.Random.nextInt(0, 10000)
+        val floatArr = FloatArray(size)
+
+        for(i in 0..size-1) {
+            floatArr[i] = kotlin.random.Random.nextDouble(-10000.0, 10000.0).toFloat()
+        }
+
+        val floatBuf = BufferUtils.allocateFloatAndPut(floatArr)
+
+        assertEquals(size, floatBuf.capacity(), "Float buffer capacity was expected to be $size, but is ${floatBuf.capacity()}")
+        assertEquals (ByteOrder.nativeOrder(), floatBuf.order(), "Float buffer was expected to be in ${ByteOrder.nativeOrder()}, but is ${floatBuf.order()}")
+
+        var index = 0
+
+       //numCorrect records how many elements in the array and buffer match
+        var numCorrect = 0
+        floatArr.forEach {
+
+            if(it == floatBuf[index])
```

Comment:
Don't do that. Always use brackets, this shorthand notation might introduce bugs that are very difficult to find.

Reply:
Also, instead of if, I'd use `assertEquals` here.

Reply:
I intentionally didn't use assertEquals to minimize the printed output. If the array contains many elements and there is something wrong with the code, there will be a print statement for each of them. What do you think? Should I use assertEquals without a message?

Reply:
You can use assertEquals with a message, because the test will bail out at the first failed assertion 👍